go/src/koding/kites/kloud/metrics/metrics.go

```diff
@@ -55,9 +56,17 @@ func (p *Publisher) Publish(r *kite.Request) (interface{}, error) {
 
 	for _, data := range req.Data {
 		_, err := p.conn.Write(data)
-		if err != nil {
-			return nil, err
+		if err == nil {
```

Comment:
if error is nil, no need to do a type casting. This check is in a critical path and most of the time it will be nil (in a healthy environment)